pkg/scheduler/framework/v1alpha1/interface.go

```diff
@@ -114,9 +114,48 @@ type PodInfo struct {
 	Timestamp time.Time
 }
 
+// SchedulingQueue is an interface for a queue to store pods waiting to be scheduled.
+// The interface follows a pattern similar to cache.FIFO and cache.Heap and
+// makes it easy to use those data structures as a SchedulingQueue.
+type SchedulingQueue interface {
```

Comment:
Once we add something to the framework, plugins may start using it and it will be hard to remove or change it in the future. As a result, I would prefer to start with a limited functionality that is needed now and add more functionality later when there is a use-case.
For that reason, I would prefer the SchedulingQueue interface that plugins can use to be a much more limited version of the actual scheduling queue. For example, I don't think we should allow pluings to `Pop`, `Close`, .... The only methods that I would expose for now is: `MoveAllToActiveQueue`, `NominatedPodsForNode`. We can add more in the future.

Reply:
So with this PR `NewFramework` would need to get a handle for the scheduling queue, and currently `NewSchedulingQueue` needs a handle to the framework. As of now `SchedulingQueue` is in package `framework`, so there are no cyclic imports. If  we keep it in `internalqueue` and have a restricted view of it in `framework`, then there are three solutions I can think of to prevent cyclic imports:
- have a `SetSchedulingQueue` method on framework so we can create the scheduling queue after constructing framework, introduces mutability so not great.
- move the `SchedulingQueue` interface some other package, maybe api, so both framework and internalqueue can freely import it
- keep both the `SchedulingQueue` interface and the interface that will be passed with `FrameworkHandle` on package `framework`. Not great as we need to expose both interfaces from `Framework`

pushed a change to do the third, but open for suggestions